### ISSUE TITLE
About the training code.

### ISSUE THREAD

**Issue by Zhentao-Liu** [HJ-harry/DDS#3]
Would you release the training codes/sripts? It seems like there is only inference code (main.py). Thanks in advance.

**Comment by Zhentao-Liu** [comment]
Data preprocessing. In your provided CT data, it has the intensity range of [0.0777, 1]. It must have some normalization. Typical CT image would not have such attenuation range. So, how do you preprocess the intensity of your data?

**Comment by Zhentao-Liu** [comment]
If I want to perform different views settings SVCT (e.g., 10/20 views,) or on different dataset (dental), should I retrain the diffusion model?

**Comment by Zhentao-Liu** [comment]
And if I change the rotation angle (360, a full scan), should I retrain the model?

**Comment by Zhentao-Liu** [comment]
Another question is that you use parallel geometry (fan-beam) to perform projection and reconstruction in SVCT. Does that meaningful? For 3D CT data, we commonly use Cone-beam geometry to perform projection and reconstruction.

**Comment by Zhentao-Liu** [comment]
` 
def Acg_TV(x):      
       return A_funcs.AT(A_funcs.A(x)) + rho * _DzT(_Dz(x))
def _Dz(x): # Batch direction
    y = torch.zeros_like(x)
    y[:-1] = x[1:]
    y[-1] = x[0]
    return y - x
def _DzT(x): # Batch direction  D H W， 沿着D这个维度
    y = torch.zeros_like(x)
    y[:-1] = x[1:]
    y[-1] = x[0]

    tempt = -(y-x)
    difft = tempt[:-1]
    y[1:] = difft
    y[0] = x[-1] - x[0]

    return y
`

I am confusing why use _DzT(_Dz(x)), it seems that _DzT is not the inverse operation of _Dz.
ADMM with ACV_TV: you want the generated data is the same with the original data in projection direction.
And z axis should be smooth.

**Comment by Edward-HE** [comment]
> Data preprocessing. In your provided CT data, it has the intensity range of [0.0777, 1]. It must have some normalization. Typical CT image would not have such attenuation range. So, how do you preprocess the intensity of your data?

Simply use standard normalization techniques common in CT image reconstruction research. For example:

- Apply window-clip and normalization to the original HU values.
- Directly normalize the HU values.

Just choose the one that best suits your specific analysis needs.

**Comment by Edward-HE** [comment]
> Another question is that you use parallel geometry (fan-beam) to perform projection and reconstruction in SVCT. Does that meaningful? For 3D CT data, we commonly use Cone-beam geometry to perform projection and reconstruction.

In this series of works, using different SVCT settings and acquisition geometries does not require retraining the diffusion model, as these are unrelated to the model's training process. As far as I know, most of Hyungjin's work uses unconditional diffusion models, which are solely trained to learn the distribution of clean medical images. He focuses on designing advanced and efficient methods to incorporate measurement data into diffusion sampling. 

Therefore, you can freely modify any settings related to the measurement data, as long as the expected reconstruction $\mathbf{x}$ lies on the manifold learned by the diffusion model.

**Comment by BanyeDguduyue** [comment]
I meet this problem like you. do u solve this problem？
